src/draw/sw/lv_draw_sw.c

```diff
@@ -137,32 +148,34 @@ static int32_t lv_draw_sw_delete(lv_draw_unit_t * draw_unit)
 
 void lv_draw_sw_rgb565_swap(void * buf, uint32_t buf_size_px)
 {
-    uint32_t u32_cnt = buf_size_px / 2;
-    uint16_t * buf16 = buf;
-    uint32_t * buf32 = buf;
-
-    while(u32_cnt >= 8) {
-        buf32[0] = ((buf32[0] & 0xff00ff00) >> 8) | ((buf32[0] & 0x00ff00ff) << 8);
-        buf32[1] = ((buf32[1] & 0xff00ff00) >> 8) | ((buf32[1] & 0x00ff00ff) << 8);
-        buf32[2] = ((buf32[2] & 0xff00ff00) >> 8) | ((buf32[2] & 0x00ff00ff) << 8);
-        buf32[3] = ((buf32[3] & 0xff00ff00) >> 8) | ((buf32[3] & 0x00ff00ff) << 8);
-        buf32[4] = ((buf32[4] & 0xff00ff00) >> 8) | ((buf32[4] & 0x00ff00ff) << 8);
-        buf32[5] = ((buf32[5] & 0xff00ff00) >> 8) | ((buf32[5] & 0x00ff00ff) << 8);
-        buf32[6] = ((buf32[6] & 0xff00ff00) >> 8) | ((buf32[6] & 0x00ff00ff) << 8);
-        buf32[7] = ((buf32[7] & 0xff00ff00) >> 8) | ((buf32[7] & 0x00ff00ff) << 8);
-        buf32 += 8;
-        u32_cnt -= 8;
-    }
+    if(LV_RESULT_INVALID == LV_DRAW_SW_RGB565_SWAP(buf, buf_size_px)) {
```

Comment:
```suggestion
    if(LV_DRAW_SW_RGB565_SWAP(buf, buf_size_px) == LV_RESULT_OK) return;
```
to lower the level of nesting